AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -414,19 +414,30 @@ - (void)setLogUrl:(NSString *)logUrl {
 
 - (void)setMaxStorageSize:(long)sizeInBytes completionHandler:(void (^)(BOOL))
 completionHandler {
-  if (self.configuredFromApplication) {
-    MSLogWarning([MSAppCenter logTag],
-        @"Unable to set storage size after the application has configured App"
-        "Center");
-    if (completionHandler) {
-      completionHandler(NO);
+  @synchronized (self) {
+    if (self.setMaxStorageSizeHasBeenCalled) {
+      MSLogWarning([MSAppCenter logTag], @"setMaxStorageSize:completionHandler: may only be called once per app "
+                                         "launch");
+      if (completionHandler) {
+        completionHandler(NO);
```

Comment:
We can run into deadlocks by allowing the completion handler to get called inside of a lock (@clpolet fixed an issue a while back where that was the cause.) Can we move that code? Or call it on another thread?

Reply:
I guess since there's just one lock, it *shouldn't* deadlock (but there could be hidden interactions with another lock elsewhere in the code), but even so, if they do something time-consuming in the callback, then all other app center operations will be blocked for that time

Reply:
Or it can at least cause App Center APIs to take a long time

Reply:
The `completionHandler` can surely be called at the end after the synchronized if we bring up some kind of success state.